### PR TITLE
wasi-preview2 article

### PR DESCRIPTION
**Pull request by ricochet**
Cross-posting from Dan's personal blog with his permission.

https://blog.sunfishcode.online/wasi-preview2/

### CODE REVIEW

**Comment by disquisitioner** [comment]
Great to have an article that highlights approval of WASI Preview 2, and I like the idea of cross-posting from Dan's blog.  (Thanks, @sunfishcode!).   Two questions:

1. Is there anything we can link to that represents the official approval vote outcome from the WASI Subgroup?
2. Is there an announcement (or something comparable to Dan's blog) about having completed the Component Model work?  (That seems worth celebrating/announcing too, and though I wouldn't hold up this post for a post there I do think it'd be good to highlight that too if possible and link to it).

**Comment by disquisitioner** [comment]
Also, per our conversation this morning @ricochet, if you want to control how the article excerpt gets generated and included in the overall News page you can add the following to the post's markup file:

```
excerpt_separator: <!--end_excerpt-->
---
This is the excerpt paragraph, generally the very first one in the post, which serves both as the
introductory paragraph in the post and will be pulled out and used to preview/tease the article
in the overall News page at https://bytecodealliance.org/articles/ along with title, posted date,
and the author's avatar.
<!--end_excerpt-->
```

**Comment by pchickey** [comment]
https://github.com/WebAssembly/WASI/pull/577 can be used as a place where the "official" approval was recorded

**Review by disquisitioner — COMMENTED** [review]
I suggest a tweak to the intro part of this so it looks like the following:

```
 ---
title: WASI Preview 2 Launched
author: "Dan Gohman"
github_name: sunfishcode
excerpt_separator: <!--end_excerpt-->
---
The WASI Subgroup voted on January 25, 2024 to launch WASI Preview 2!
This blog post is a brief look at the present, past, and future of WASI,
and a repost of an original article shared
[here]( https://blog.sunfishcode.online/wasi-preview2/).
<!--end_excerpt-->

## The present
```

This yields a reasonable intro that can also serve as the teaser "excerpt" to show on the overall News page of Bytecode Alliance website and also links to Dan's original blog post.

Everything else looks fine!

**Comment by sunfishcode** [comment]
Sounds good to me!

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
title: WASI 0.2 Launched
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
The WASI Subgroup has just voted to launch WASI 0.2, also known as WASI Preview 2, or WASIp2! This blog post
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
The Subgroup voted to launch WASI 0.2!
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
says that the WASI 0.2 APIs are stable. There is still a lot more
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
WASI 0.2 includes two *worlds*:
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
And so, WASI 0.2 represents not just a new API, but a new approach to
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
worry so much about getting WASI 0.2 itself perfectly right for all time,
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
This is why the Subgroup voting to launch WASI 0.2 is such a big deal.
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
because we know we can virtualize WASI 0.2 itself in terms of future APIs.
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
looking forward to. With WASI 0.2.0 finalized, it'll be time to start planning
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
for things we want to add in a WASI 0.2.1 soon after it. That'll be another
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
milestone: the first update to WASI 0.2, which will establish how to do updates,
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
and set up the rhythm for doing incremental updates. WASI 0.2 is small still;
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
looking forward to doing many updates in WASI 0.2.1 and beyond.
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
And when we do get there, the transition from WASI 0.2 to WASI 0.3 should be much smoother
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
than WASI 0.1 to WASI 0.2. The component model's virtualizability means it
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
easier for engines to support both WASI 0.2 and WASI 0.3 at the same time if
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
should be easier to polyfill WASI 0.2 in terms of WASI 0.3. And, it should be
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
they wish to, because WASI 0.2 will essentially use a subset of WASI 0.3's
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
WASI 0.2, also known as WASI Preview 2 or WASIp2, is a major milestone that many people have contributed to, and
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
At the same time, work towards WASI 0.3, also known as Preview 3, will be getting underway. The
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
major banner of WASI 0.3 is *async*, and adding the `future` and `stream` types
```

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
will take some time, so while the design work is starting now, WASI 0.3
```

**Review by sunfishcode — APPROVED** [review]
I've heard feedback that calling it "WASI 0.2" instead of "WASI Preview 2" is easier to understand. If y'all agree, I've made these edits as review suggestions here. Either way, this looks good to me!

File: _posts/2024-01-25-WASI-Preview2.md
Comment (disquisitioner) [reply _posts/2024-01-25-WASI-Preview2.md]: I'm OK with "WASI 0.2", especially as we consistently used that terminology during last week's Plumber's Summit.  I do want to make sure we let people know early on in the post that what we are calling "WASI 0.2" here used to be widely referred to "WASI Preview 2" and is still going to be encountered using that previous nomenclature. (E.g. the repo is WASI/preview2.)

So I'm fine with WASI 0.2 in the title but think we should make that point as early in the document as possible to minimize confusion.

Also, Eric Gregory has just staged a PR with his write-ups from both days of last week's Plumber's Summit and he uses the "Preview 2" and "Preview 3" nomenclature there.  Should we ask him to change that??

Lastly, you aren't modifying your blog post to be "0.2" instead of "Preview 2" are you @sunfishcode ??

File: _posts/2024-01-25-WASI-Preview2.md
Comment (disquisitioner) [reply _posts/2024-01-25-WASI-Preview2.md]: Can we put the "formerly known as 'Preview 2'" clarification here as the opening sentence in the post and the first opportunity to do so (rather than as the next change below)?

File: _posts/2024-01-25-WASI-Preview2.md
Comment (disquisitioner) [reply _posts/2024-01-25-WASI-Preview2.md]: If this is the best place to call out that 0.2 was formerly known as Preview 2 then OK, but I proposed doing that in the opening paragraph in my comment above.  Let me know what you think.

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [reply _posts/2024-01-25-WASI-Preview2.md]: > So I'm fine with WASI 0.2 in the title but think we should make that point as early in the document as possible.

Yes, on line 13 here I have added a mention that "WASI 0.2" is also known as "WASI Preview 2".

> Also, Eric Gregory has just staged a PR with his write-ups from both days of last week's Plumber's Summit and he uses the "Preview 2" and "Preview 3" nomenclature there. Should we ask him to change that??

Yeah. It's not absolutely necessary, but it would be good if he could change that.

> Lastly, you aren't modifying your blog post to be "0.2" instead of "Preview 2" are you @sunfishcode ??

I have now modified the post to be "0.2".

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [reply _posts/2024-01-25-WASI-Preview2.md]: Sure.

File: _posts/2024-01-25-WASI-Preview2.md
Comment (disquisitioner) [reply _posts/2024-01-25-WASI-Preview2.md]: Cool. I relayed word to @ericgregory so he can look at changing things in his posts.

Also, will the URL to your post still be https://blog.sunfishcode.online/wasi-preview2/?

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [reply _posts/2024-01-25-WASI-Preview2.md]: Sounds good. I've now updated the edits.

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [reply _posts/2024-01-25-WASI-Preview2.md]: I'll keep that URL live, but I'll add a new URL based on WASI 0.2 as soon as I can figure out how to do that in my blog software here.

Comment (sunfishcode) [reply _posts/2024-01-25-WASI-Preview2.md]: The new URL is now live: https://blog.sunfishcode.online/wasi-0-2/

File: _posts/2024-01-25-WASI-Preview2.md
Comment (sunfishcode) [inline _posts/2024-01-25-WASI-Preview2.md]:
```suggestion
> Original post from https://blog.sunfishcode.online/wasi-0-2/
```

**Comment by sunfishcode** [comment]
I could also submit a PR to this branch if you'd like, if that'd be easier than all these suggested changes.

**Comment by disquisitioner** [comment]
An updated PR in this branch with all these changes would be great, @sunfishcode -- thanks!  That way I can wear the reviewer hat plus also still test the branch stand-alone on my development machine to check URLs, etc.

**Comment by ricochet** [comment]
Hey I welcome someone taking this over. I should have time to pull in edits this weekend but I'm spread a little too thin for the next few days.

**Comment by sunfishcode** [comment]
@disquisitioner I've now posted https://github.com/bytecodealliance/bytecodealliance.org/pull/85 with a branch with the WASI 0.2 changes.

**Comment by disquisitioner** [comment]
Merging into gh-pages now that we've incorporated additional updates from @sunfishcode's branch.

**Pull request merged**
